src/odemis/gui/comp/overlay/polygon.py

```diff
@@ -0,0 +1,225 @@
+
+    def __init__(self, cnvs, colour=gui.SELECTION_COLOUR):
+        """
+        :param: cnvs: canvas for the overlay.
+        :param: colour (str): hex colour code for the polygon.
+        """
+        WorldOverlay.__init__(self, cnvs)
+        LineEditingMixin.__init__(self, colour)
+        EditableShape.__init__(self, cnvs)
+
+        self._label = self.add_label("", colour=self.colour, align=wx.ALIGN_CENTRE_HORIZONTAL)
+        self.v_point.subscribe(self._on_v_point)
+
+    def _on_v_point(self, point):
+        """Callback for a new value v_point in ClickMixin."""
+        offset = self.cnvs.get_half_buffer_size()
+        p_point = Vec(self.cnvs.view_to_phys(point, offset))
+        self._points.append(p_point)
+
+    def _view_to_phys(self):
+        if len(self.v_points) == len(self._points):
+            offset = self.cnvs.get_half_buffer_size()
+            for idx, point in enumerate(self.v_points):
+                self._points[idx] = Vec(self.cnvs.view_to_phys(point, offset))
+
+    def _phys_to_view(self):
+        if len(self._points) == len(self.v_points):
+            offset = self.cnvs.get_half_buffer_size()
+            for idx, point in enumerate(self._points):
+                self.v_points[idx] = Vec(self.cnvs.phys_to_view(point, offset))
+
+    def is_point_in_shape(self, point):
+        # A polygon should have atleast 2 points after on_right_up
+        if len(self._points) > 2:
+            return point_in_polygon(point, self._points)
+        return False
+
+    def on_left_down(self, evt):
+        if self.active.value and self.selected.value:
+            LineEditingMixin._on_left_down(self, evt)
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_left_down(self, evt)
+
+    def on_left_up(self, evt):
+        if self.active.value:
+            LineEditingMixin._on_left_up(self, evt)
+            if self.finished_click_mixin:
+                self._phys_to_view()
+                offset = self.cnvs.get_half_buffer_size()
+                p_point = Vec(self.cnvs.view_to_phys(evt.Position, offset))
+                selected = self.is_point_in_shape(p_point)
+                self.selected._set_value(selected, must_notify=True)
+            self.cnvs.update_drawing()
+        WorldOverlay.on_left_up(self, evt)
+
+    def on_right_down(self, evt):
+        if self.active.value:
+            LineEditingMixin._on_right_down(self, evt)
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_right_down(self, evt)
+
+    def on_right_up(self, evt):
+        if self.active.value:
+            LineEditingMixin._on_right_up(self, evt)
+            self._view_to_phys()
+            if len(self._points) <= 2:
+                logging.warning("Cannot create a polygon for less than 3 points.")
+                self.reset_click_mixin()
+                self._points.clear()
+            # FastEMROA.get_poly_field_indices expects list of nested tuples (y, x)
```

Comment:
That's really confusing that the points are in `(y,x)`. I guess it comes from numpy somewhere? Is there a really really good reason, or can we change `get_poly_field_indices` to follow the standard convention x,y?